nilearn/maskers/nifti_labels_masker.py

```diff
@@ -701,7 +679,7 @@ def transform_single_imgs(self, imgs, confounds=None, sample_mask=None):
 
         if self.labels is not None:
             self.region_names_ = {
-                key: self.labels[region_id]
+                key: self.labels[self.labels_.index(region_id)]
```

Comment:
This is the important bit that will let your use the label maskers with all atlases but the output region names is sometimes messed up when you have a background listed as a label.

Reply:
We probably want to do the "background" check in a case insensitive manner in case some atlas decide to go for `background`, or `BackGround` (or heaven forbids `BaCkGrOuNd`).

Also users of other atlases (not just "ours") may be in trouble if their labels have some entirely different word for backgrund or some abbreviation (bckgrd, bg...). So this is a "big" hard coded assumption we should mention in the doc string if it is not there already.